lib/BoundaryValueDiffEqMIRKN/src/mirkn.jl

```diff
@@ -96,71 +96,84 @@ function SciMLBase.solve!(cache::MIRKNCache{iip, T}) where {iip, T}
     nlsolve_alg = __concrete_nonlinearsolve_algorithm(nlprob, cache.alg.nlsolve)
     sol_nlprob = __solve(nlprob, nlsolve_alg; kwargs..., alias_u0 = true)
     recursive_unflatten!(cache.y₀, sol_nlprob.u)
-    solu = ArrayPartition.(cache.y₀[1:length(mesh)], cache.y₀[(length(mesh) + 1):end])
+    L = length(mesh)
+    solu = ArrayPartition.(cache.y₀.u[1:L], cache.y₀.u[(L + 1):end])
     return SciMLBase.build_solution(
         prob, cache.alg, mesh, solu; retcode = sol_nlprob.retcode)
 end
 
-function __construct_nlproblem(cache::MIRKNCache{iip}, y::AbstractVector) where {iip}
+function __construct_nlproblem(cache::MIRKNCache{iip}, y) where {iip}
     (; alg) = cache
     pt = cache.problem_type
+
     loss = if iip
         @closure (du, u, p) -> __mirkn_loss!(
             du, u, p, cache.y, pt, cache.bc, cache.residual, cache.mesh, cache)
     else
         @closure (u, p) -> __mirkn_loss(u, p, cache.y, pt, cache.bc, cache.mesh, cache)
     end
 
-    lossₚ = (iip ? __Fix3 : Base.Fix2)(loss, cache.p)
-    sd = alg.jac_alg.diffmode isa AutoSparse ? SymbolicsSparsityDetection() :
-         NoSparsityDetection()
-    ad = alg.jac_alg.diffmode
-    lz = reduce(vcat, cache.y₀)
-    jac_cache = __sparse_jacobian_cache(Val(iip), ad, sd, lossₚ, lz, lz)
-    jac_prototype = init_jacobian(jac_cache)
+    diffmode = if alg.jac_alg.diffmode isa AutoSparse
+        AutoSparse(get_dense_ad(alg.jac_alg.diffmode);
+            # Local because iszero require primal values
```

Comment:
This means that the sparsity pattern will not necessarily generalize across input points. To increase robustness, you probably want to evaluate it at `x = rand(n)` rather than, say, `x = zeros(n)`.